managed/src/main/java/com/yugabyte/yw/models/HealthCheck.java

```diff
@@ -36,15 +60,18 @@
 
   // The Json serialized version of the details. This is used only in read from and writing to the
   // DB.
+  @DbJson
   @Constraints.Required
   @Column(columnDefinition = "TEXT", nullable = false)
-  public String detailsJson;
+  public details detailsJson;
 
   public boolean hasError() {
-    JsonNode details = Json.parse(detailsJson);
-    JsonNode hasErrorField = details.get(FIELD_HAS_ERROR);
-    // Only return true if we have the top-level has_error field with a value of true.
-    return hasErrorField != null && hasErrorField.asBoolean();
+    try {
+      // Only return true if we have the top-level has_error field with a value of true.
+      return detailsJson.has_error;
+    } catch (NullPointerException e) {
```

Comment:
why npe catch? was that temporary to debug failing test?
If so remove. Otherwise use if condition to get rid of NPE.